sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/LookupCatalog.scala

```diff
@@ -26,8 +28,17 @@ import org.apache.spark.sql.catalyst.TableIdentifier
 @Experimental
 trait LookupCatalog {
 
+  protected def defaultCatalogName: Option[String] = None
   protected def lookupCatalog(name: String): CatalogPlugin
 
+  lazy val defaultCatalog: Option[CatalogPlugin] = {
+    defaultCatalogName.flatMap(name => Try(lookupCatalog(name)).toOption)
```

Comment:
Is it ok to suppress other exceptions than `CatalogNotFoundException`?
Even for CatalogNotFoundException, how can we warn the user either default catalog name or default catalog plugin is misconfigured .

Reply:
No, this should log an error.

Reply:
I fixed the use of `Try`. Now these log errors instead of ignoring them.